core/tests/ledger_cleanup.rs

```diff
@@ -260,18 +270,28 @@ mod tests {
         let exit_cpu = Arc::new(AtomicBool::new(false));
         let sys = CpuStatsUpdater::new(&exit_cpu);
 
-        let mut generated_batches = VecDeque::<Vec<Shred>>::new();
+        let shreds: Arc<Mutex<VecDeque<Vec<Shred>>>> = Arc::new(Mutex::new(VecDeque::new()));
 
+        let mut entries_batch = make_many_slot_entries(0, batch_size, entries_per_slot).0;
         if pre_generate_data {
             let t0 = Instant::now();
             eprintln!("PRE_GENERATE_DATA: (this may take a while)");
             for i in 0..batches {
                 let start_slot = i * batch_size;
-                let (shreds, _) = make_many_slot_entries(start_slot, batch_size, entries_per_slot);
-                generated_batches.push_back(shreds);
+                let (new_shreds, _) = make_many_slot_entries(
+                    start_slot, batch_size, entries_per_slot);
+                shreds.lock().unwrap().push_back(new_shreds);
```

Comment:
nit: This isn't in the timed section of the test, but you could create a regular `VecDeque` to start, push all the shreds, and then move it into an `Arc<Mutex< >>` to avoid repeatedly grabbing the lock in the loop

Reply:
Let me see if I can just grab the lock once and put a bunch.   Originally it was first put in the dequeue and move it to the shared queue in run time, but I found the main thread is still not fast enough to serve the writers so I instead directly put the pre-generated slots into the shared queue.

Reply:
No yeah, I think your refactor to put all of the data into the shared container at the start is a good one. I was just saying that you can fill the container up first, and then make it an `Arc<Mutex< >>`, something like:
```
let shreds = VecDeque::new();
let mut entries_batch = make_many_slot_entries(0, batch_size, entries_per_slot).0;
if pre_generate_data {
    let t0 = Instant::now();
    eprintln!("PRE_GENERATE_DATA: (this may take a while)");
    for i in 0..batches {
        let start_slot = i * batch_size;
        let (new_shreds, _) = make_many_slot_entries(
            start_slot, batch_size, entries_per_slot);
        shreds.push_back(new_shreds);
    }
}
let shreds = Arc::new(Mutex::new(shreds));
```

Reply:
Aghh got it!  Thanks for the Rust tips!  I am still learning and might often time missing something or do it incorrectly.